packages/ui-truncate-text/src/TruncateText/index.tsx

```diff
@@ -62,56 +64,55 @@ class TruncateText extends Component<TruncateTextProps> {
     truncate: 'character',
     position: 'end',
     ignore: [' ', '.', ','],
-    debounce: 0,
-    // @ts-expect-error ts-migrate(6133) FIXME: 'truncated' is declared but its value is never rea... Remove this comment to see the full error message
-    onUpdate: (truncated, text) => {}
+    debounce: 0
   }
 
   ref: Element | null = null
+  private _text?: JSX.Element
+  private _debounced?: Debounced
+  private _stage: HTMLSpanElement | null = null
+  private _wasTruncated?: boolean
+  private _resizeListener?: ResizeObserver
 
-  // @ts-expect-error ts-migrate(7006) FIXME: Parameter 'props' implicitly has an 'any' type.
-  constructor(props) {
+  constructor(props: TruncateTextProps) {
     super(props)
     this.state = this.initialState
   }
+
   get _ref() {
     console.warn(
       '_ref property is deprecated and will be removed in v9, please use ref instead'
     )
-
     return this.ref
   }
 
   get initialState() {
     return {
       isTruncated: false,
       needsSecondRender: true,
-      truncatedElement: null,
-      truncatedText: null
+      truncatedElement: undefined,
+      truncatedText: undefined
     }
   }
 
   componentDidMount() {
     const { children, makeStyles } = this.props
 
-    // @ts-expect-error ts-migrate(2722) FIXME: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
-    makeStyles()
+    makeStyles?.()
 
     if (children) {
       this.checkChildren()
-      // @ts-expect-error ts-migrate(2339) FIXME: Property '_text' does not exist on type 'TruncateT... Remove this comment to see the full error message
-      this._text = ensureSingleChild(children)
+      const txt = ensureSingleChild(children)
+      this._text = txt ? txt : undefined
```

Comment:
this new code is needed because I wanted to use `undefined` everywhere for the lack of value instead of the mix of `undefined` and `null`